<launch>
  <arg name="camera_frame" default="/head_camera_rgb_optical_frame" />

  <arg name="sim" />
  <arg if="$(arg sim)" name="point_cloud_topic" default="/mock_point_cloud" />
  <arg if="$(arg sim)" name="depth_topic" default="/head_camera/depth_registered/image_raw" />
  <arg if="$(arg sim)" name="color_topic" default="/head_camera/rgb/image_raw" />

  <arg unless="$(arg sim)" name="point_cloud_topic" default="/head_mount_kinect/depth_registered/points" />
  <arg unless="$(arg sim)" name="depth_topic" default="/head_camera/depth_registered/image_raw" />
  <arg unless="$(arg sim)" name="color_topic" default="/head_camera/rgb/image_raw" />

  <group ns="rapid_pbd"> 
    <param name="robot" value="fetch" />
    <node pkg="rapid_pbd" type="fetch_action_node" name="fetch_actuator_server" />
    <node pkg="rapid_pbd" type="program_execution_node" name="program_executor">
      <remap from="arm_joint_action" to="/arm_controller/follow_joint_trajectory" />
      <remap from="gripper_action" to="/gripper_controller/gripper_action" />
      <remap from="head_action" to="/head_controller/follow_joint_trajectory" />
      <remap from="move_group" to="/move_group" />
    </node>
    <node pkg="rapid_pbd" type="surface_segmentation_node" name="surface_segmentation_node" args="$(arg point_cloud_topic)" />
  </group>

  <!-- Depth cloud -->
  <node pkg="nodelet" type="nodelet" name="rapid_pbd_convert_metric" args="standalone depth_image_proc/convert_metric" if="$(arg sim)">
    <remap from="image_raw" to="$(arg depth_topic)" />
    <remap from="image" to="$(arg depth_topic)_float" />
  </node>
  <node pkg="depthcloud_encoder" type="depthcloud_encoder_node" name="rapid_pbd_depthcloud_encoder">
    <param name="rgb" value="$(arg color_topic)" />
    <param if="$(arg sim)" name="depth" value="$(arg depth_topic)" />
    <param unless="$(arg sim)" name="depth" value="$(arg depth_topic)_float" />
    <param name="camera_frame_id" value="$(arg camera_frame)" />
  </node>
</launch>
